### PR TITLE
Create a doc on how Grid configs work 

### PR DESCRIPTION
**Pull request by aracho1**
Co-authored by: @andrew-nowak

Documentation on how Grid configs work and which files should be used for which purposes

### CODE REVIEW

File: docs/06-objects-of-interest/02-config.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+#### [`application.conf`](https://github.com/guardian/grid/blob/3fb1d936f82a0cd02a2a3c4293b680f7f8f5caf5/common-lib/src/main/resources/application.conf)
+
+The file lists configs that every user and service should have. 
+
+#### `common.conf` in S3
+Anything that's specific to the Guardian and to a stage, but is common across all the services.
+
+#### `kahuna.conf` in S3
+Kahuna-specific configs (one for each stage)
```
Comment (andrew-nowak) [inline docs/06-objects-of-interest/02-config.md:14]:
```suggestion
#### `<configRoot>/<stage>/<service>.conf`
Service-specific configs. These will override all other config files.
```

File: docs/06-objects-of-interest/02-config.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# Config
+
+The Grid has different config files depending on the service and usage e.g. the Guardian or BBC-specific features.
+[`GridConfigLoader.scala`](https://github.com/guardian/grid/blob/main/common-lib/src/main/scala/com/gu/mediaservice/lib/config/GridConfigLoader.scala) loads the configs, which get parsed and turned into a tree of config by a function in this library (https://github.com/lightbend/config) which will recursively merge the trees.
+
+#### [`application.conf`](https://github.com/guardian/grid/blob/3fb1d936f82a0cd02a2a3c4293b680f7f8f5caf5/common-lib/src/main/resources/application.conf)
+
+The file lists configs that every user and service should have. 
+
+#### `common.conf` in S3
```
Comment (andrew-nowak) [inline docs/06-objects-of-interest/02-config.md:10]:
```suggestion
#### `<configRoot>/<stage>/common.conf`
```

File: docs/06-objects-of-interest/02-config.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
+# Config
+
+The Grid has different config files depending on the service and usage e.g. the Guardian or BBC-specific features.
+[`GridConfigLoader.scala`](https://github.com/guardian/grid/blob/main/common-lib/src/main/scala/com/gu/mediaservice/lib/config/GridConfigLoader.scala) loads the configs, which get parsed and turned into a tree of config by a function in this library (https://github.com/lightbend/config) which will recursively merge the trees.
+
+#### [`application.conf`](https://github.com/guardian/grid/blob/3fb1d936f82a0cd02a2a3c4293b680f7f8f5caf5/common-lib/src/main/resources/application.conf)
+
+The file lists configs that every user and service should have.
```
Comment (andrew-nowak) [inline docs/06-objects-of-interest/02-config.md:8]:
```suggestion
The file lists configs that every user and service should have. This file should provide good defaults for grid features where required. This file has the lowest priority and keys will be overridden if set in other files.

This file will be packaged by sbt-native-packager, available once installed on the destination instance and automatically loaded by Play.

> **Note**
> Currently this is the only file that is loaded at Play's first initialisation. Some play settings will only take effect if set in this file.
```

File: docs/06-objects-of-interest/02-config.md [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+The Grid has different config files depending on the service and usage e.g. the Guardian or BBC-specific features.
+[`GridConfigLoader.scala`](https://github.com/guardian/grid/blob/main/common-lib/src/main/scala/com/gu/mediaservice/lib/config/GridConfigLoader.scala) loads the configs, which get parsed and turned into a tree of config by a function in this library (https://github.com/lightbend/config) which will recursively merge the trees.
+
+#### [`application.conf`](https://github.com/guardian/grid/blob/3fb1d936f82a0cd02a2a3c4293b680f7f8f5caf5/common-lib/src/main/resources/application.conf)
+
+The file lists configs that every user and service should have. 
+
+#### `common.conf` in S3
+Anything that's specific to the Guardian and to a stage, but is common across all the services.
```
Comment (andrew-nowak) [inline docs/06-objects-of-interest/02-config.md:11]:
```suggestion
Anything that's specific to an organisation and to a stage, but is common across all the services. Keys set here will override those set in `application.conf` but be overridden by the service-specific configs.
```

**Review by dskamiotis — APPROVED** [review]
Looks good from my perspective (although not a Grid expert)
Really handy :)

File: docs/06-objects-of-interest/02-config.md [hunk 5]
```diff
@@ -0,0 +1,14 @@
+# Config
+
+The Grid has different config files depending on the service and usage e.g. the Guardian or BBC-specific features.
+[`GridConfigLoader.scala`](https://github.com/guardian/grid/blob/main/common-lib/src/main/scala/com/gu/mediaservice/lib/config/GridConfigLoader.scala) loads the configs, which get parsed and turned into a tree of config by a function in this library (https://github.com/lightbend/config) which will recursively merge the trees.
+
+#### [`application.conf`](https://github.com/guardian/grid/blob/3fb1d936f82a0cd02a2a3c4293b680f7f8f5caf5/common-lib/src/main/resources/application.conf)
+
+The file lists configs that every user and service should have. 
+
```
Comment (andrew-nowak) [inline docs/06-objects-of-interest/02-config.md:9]:
```suggestion

### Other config files

Other config files will be read from `$HOME/.grid/` (when running locally) or `/etc/grid` (when deployed, as determined by the contents of `/etc/grid/stage`).
For local development, files will be generated by the [generate-config script](/dev/script/generate-config). It is up to the implementor to determine how to load config files onto deployed instances - we at the Guardian currently fetch files from S3 in a [UserData script](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/user-data.html).
 
```

**Review by andrew-nowak — COMMENTED** [review]
looks really good, thanks for getting this started! I've filled out with suggestions for a few more details and tried to make it less guardian-specific

**Pull request merged**